src/command.ts

```diff
@@ -24,8 +24,17 @@ const isAdmin = (ctx: Context) =>
         .getAuthor()
         .then((author) => ["administrator", "creator"].includes(author.status));
 
-export const matchesPattern = (value: string, pattern: string | RegExp) =>
-    typeof pattern === "string" ? value === pattern : pattern.test(value);
+export const matchesPattern = (
+    value: string,
+    pattern: string | RegExp,
+    ignoreCase = false,
+) => {
+    const transformedValue = ignoreCase ? value.toLowerCase() : value;
+
+    return typeof pattern === "string"
```

Comment:
```suggestion
    const transformedValue = ignoreCase ? value.toLowerCase() : value;
    if (typeof pattern === "object" && ignoreCase) {
        pattern = new RegExp(pattern, pattern.flags + "i");
    }
```

Reply:
Its best if we don't reassign `pattern` here. IMO

Reply:
U think it's better to reasign at command creation?

Reply:
no, I mean just create another variable so we don't reassign to the parameter